pkg/statistics/handle/cache/statscacheinner.go

```diff
@@ -92,16 +93,20 @@ func (sc *StatsCache) putCache(id int64, t *statistics.Table) bool {
 
 // Put puts the table statistics to the cache.
 func (sc *StatsCache) put(id int64, t *statistics.Table) {
-	ok := sc.putCache(id, t)
-	if !ok {
+	for {
+		// retry if the cache is full
+		ok := sc.putCache(id, t)
+		if ok {
+			// update the maxTblStatsVer
+			for v := sc.maxTblStatsVer.Load(); v < t.Version; v = sc.maxTblStatsVer.Load() {
+				if sc.maxTblStatsVer.CompareAndSwap(v, t.Version) {
+					break
+				} // other goroutines have updated the sc.maxTblStatsVer, so we need to check again.
+			}
+			return
+		}
 		logutil.BgLogger().Warn("fail to put the stats cache", zap.Int64("id", id))
```

Comment:
Do we need to modify or remove this log?
It's possible that there're too many logs when startup

Reply:
May add an mechanism to reduce the log